src/mgmt/srv_drpc.c

```diff
@@ -182,50 +183,70 @@ ds_mgmt_drpc_group_update(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 }
 
 static int
-create_pool_props(daos_prop_t **out_prop, char *owner, char *owner_grp,
-		  char *label, const char **ace_list, size_t ace_nr)
+create_pool_props_from_req(daos_prop_t **out_prop, const Mgmt__PoolCreateReq *req)
+//char *owner, char *owner_grp, char *label, const char **ace_list, size_t ace_nr, const struct policy_desc_t *policy_desc)
 {
-	char		*out_owner = NULL;
-	char		*out_owner_grp = NULL;
-	char		*out_label = NULL;
-	struct daos_acl	*out_acl = NULL;
-	daos_prop_t	*new_prop = NULL;
-	uint32_t	entries = 0;
-	uint32_t	idx = 0;
-	int		rc = 0;
+	char			*out_owner = NULL;
+	char			*out_owner_grp = NULL;
+	char			*out_label = NULL;
+	struct daos_acl		*out_acl = NULL;
+	struct policy_desc_t 	*out_policy_desc = NULL;
+	daos_prop_t		*new_prop = NULL;
+	uint32_t		entries = 0;
+	uint32_t		idx = 0;
+	int			rc = 0;
 
-	if (ace_list != NULL && ace_nr > 0) {
-		rc = daos_acl_from_strs(ace_list, ace_nr, &out_acl);
+	if (req->acl != NULL && req->n_acl > 0) {
+		rc = daos_acl_from_strs((const char **)req->acl, req->n_acl, &out_acl);
 		if (rc != 0)
 			D_GOTO(err_out, rc);
 
 		entries++;
 	}
 
-	if (owner != NULL && *owner != '\0') {
-		D_ASPRINTF(out_owner, "%s", owner);
+	if (req->user != NULL && *req->user != '\0') {
+		D_ASPRINTF(out_owner, "%s", req->user);
 		if (out_owner == NULL)
 			D_GOTO(err_out, rc = -DER_NOMEM);
 
 		entries++;
 	}
 
-	if (owner_grp != NULL && *owner_grp != '\0') {
-		D_ASPRINTF(out_owner_grp, "%s", owner_grp);
+	if (req->usergroup != NULL && *req->usergroup != '\0') {
+		D_ASPRINTF(out_owner_grp, "%s", req->usergroup);
 		if (out_owner_grp == NULL)
 			D_GOTO(err_out, rc = -DER_NOMEM);
 
 		entries++;
 	}
 
-	if (label != NULL && *label != '\0') {
-		D_ASPRINTF(out_label, "%s", label);
+	if (req->label != NULL && *req->label != '\0') {
+		D_ASPRINTF(out_label, "%s", req->label);
 		if (out_label == NULL)
 			D_GOTO(err_out, rc = -DER_NOMEM);
 
 		entries++;
 	}
 
+	D_ALLOC(out_policy_desc, sizeof(struct policy_desc_t));
+ 		if (out_policy_desc == NULL)
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs